portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,59 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
 import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("text-input");
+
+    if (StringUtils.isNotEmpty(text)) {
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("message",text);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+    }
+    
+    response.sendRedirect("/index.html");
```

Comment:
why redirect? Just return a response.

Reply:
A redirect is needed to go back to the homepage, else it stays at /data. I can't return a response as doPost() returns void.

Reply:
Return the void response. Check for the status code on the client side, and then add the comment to the comment container

Reply:
Also add some validation on the client side. I added a single space as a comment.

Reply:
An empty comment does not return an error. It just redirects to index.html

Reply:
Added client and server side validation.
Also removed redirection and loaded comments based on the response.